From d21711466f3409925d643cfa317dd0245e24839b Mon Sep 17 00:00:00 2001
From: Lorenzo Gaifas <brisvag@gmail.com>
Date: Thu, 20 Jul 2023 22:07:21 +0200
Subject: [PATCH] Bump vispy to 0.13 (#6025)

Bump vispy to 0.13 to get some extra features. Relevant changes are:

- vispy/vispy#2465, which can improve performance dramatically for
`mip`-style projections in volumes by stopping the ray cast early if the
max clim was already reached
- a couple improvements to touchaped gestures (vispy/vispy#2456,
vispy/vispy#2483)

Reverted defaults of `Markers.scaling` also mean that we need to add a
couple lines in napari to maintain the same behaviour ( see
vispy/vispy#2359 and #5312 if you want to dive deeper).

cc @aganders3 who did much of the vispy work :)

---------

Co-authored-by: Matthias Bussonnier <bussonniermatthias@gmail.com>
Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>
---
 napari/_vispy/visuals/points.py               |  13 +-
 resources/constraints/constraints_py3.10.txt  | 140 ++++++++++-------
 resources/constraints/constraints_py3.11.txt  |  67 ++++----
 resources/constraints/constraints_py3.8.txt   | 138 ++++++++++-------
 resources/constraints/constraints_py3.9.txt   | 144 +++++++++--------
 .../constraints_py3.9_examples.txt            | 146 ++++++++++--------
 resources/requirements_mypy.txt               |  14 +-
 setup.cfg                                     |   6 +-
 8 files changed, 389 insertions(+), 279 deletions(-)

diff --git a/napari/_vispy/visuals/points.py b/napari/_vispy/visuals/points.py
index 5b3eff2b..b365734c 100644
--- a/napari/_vispy/visuals/points.py
+++ b/napari/_vispy/visuals/points.py
@@ -17,7 +17,14 @@ class PointsVisual(ClippingPlanesMixin, Compound):
     """

     def __init__(self) -> None:
-        super().__init__([Markers(), Markers(), Line(), Text()])
+        super().__init__(
+            [
+                Markers(scaling='visual'),
+                Markers(scaling='visual'),
+                Line(),
+                Text(),
+            ]
+        )
         self.scaling = True

     @property
@@ -26,12 +33,12 @@ class PointsVisual(ClippingPlanesMixin, Compound):
         Scaling property for both the markers visuals. If set to true,
         the points rescale based on zoom (i.e: constant world-space size)
         """
-        return self._subvisuals[0].scaling
+        return self._subvisuals[0].scaling == 'visual'

     @scaling.setter
     def scaling(self, value):
         for marker in self._subvisuals[:2]:
-            marker.scaling = value
+            marker.scaling = 'visual' if value else 'fixed'

     @property
     def antialias(self):
diff --git a/resources/constraints/constraints_py3.10.txt b/resources/constraints/constraints_py3.10.txt
index 758c681d..a081ce95 100644
--- a/resources/constraints/constraints_py3.10.txt
+++ b/resources/constraints/constraints_py3.10.txt
@@ -20,6 +20,7 @@ attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
+    #   referencing
 babel==2.12.1
     # via
     #   napari (setup.cfg)
@@ -34,25 +35,25 @@ certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.4
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.5
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.5.0
+dask==2023.7.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -66,7 +67,7 @@ docutils==0.17.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.1
+exceptiongroup==1.1.2
     # via
     #   hypothesis
     #   pytest
@@ -74,16 +75,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.4
+fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.5.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,28 +92,28 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.75.3
+hypothesis==6.81.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.1
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.8.0
     # via dask
-in-n-out==0.1.7
+in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.1
+ipykernel==6.24.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.13.2
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -126,13 +127,15 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.17.3
+jsonschema==4.18.0
     # via napari (setup.cfg)
-jupyter-client==8.2.0
+jsonschema-specifications==2023.6.1
+    # via jsonschema
+jupyter-client==8.3.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -141,23 +144,23 @@ kiwisolver==1.4.4
     # via
     #   matplotlib
     #   vispy
-lazy-loader==0.2
+lazy-loader==0.3
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.3
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
-lxml==4.9.2
+lxml==4.9.3
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.1
+matplotlib==3.7.2
     # via napari (setup.cfg)
 matplotlib-inline==0.1.6
     # via
@@ -176,7 +179,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -185,10 +190,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.25.1
     # via
     #   contourpy
     #   dask
@@ -248,7 +255,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.3 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -260,32 +267,32 @@ pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==9.5.0
+pillow==10.0.0
     # via
     #   imageio
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.5.1
+platformdirs==3.8.1
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
     #   napari (setup.cfg)
     #   scikit-image
-prompt-toolkit==3.0.38
+prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -295,7 +302,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.11
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -309,7 +316,9 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
@@ -320,9 +329,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -340,7 +349,7 @@ pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.4.0
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -348,13 +357,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -376,7 +385,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -390,20 +399,29 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.30.0
+referencing==0.29.1
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.5
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+rpds-py==0.8.10
+    # via
+    #   jsonschema
+    #   referencing
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.10.1 ; python_version >= "3.9"
+scipy==1.11.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -444,11 +462,14 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
+    #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
-tifffile==2023.4.12
+tensorstore==0.1.40
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+tifffile==2023.7.4
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -486,11 +507,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.7.1
     # via
     #   app-model
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
@@ -498,11 +520,11 @@ typing-extensions==4.5.0
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.2
+urllib3==2.0.3
     # via requests
-virtualenv==20.23.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
-vispy==0.12.2
+vispy==0.13.0
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -518,15 +540,17 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.6.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
-zipp==3.15.0
+zipp==3.16.0
     # via importlib-metadata

 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.11.txt b/resources/constraints/constraints_py3.11.txt
index d3a85530..30092961 100644
--- a/resources/constraints/constraints_py3.11.txt
+++ b/resources/constraints/constraints_py3.11.txt
@@ -20,6 +20,7 @@ attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
+    #   referencing
 babel==2.12.1
     # via
     #   napari (setup.cfg)
@@ -34,9 +35,9 @@ certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.4
     # via
     #   dask
     #   typer
@@ -52,7 +53,7 @@ coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.6.0
+dask==2023.7.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -87,7 +88,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.79.1
+hypothesis==6.81.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -98,13 +99,13 @@ imageio==2.31.1
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.7.0
+importlib-metadata==6.8.0
     # via dask
-in-n-out==0.1.7
+in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.2
+ipykernel==6.24.0
     # via
     #   napari-console
     #   qtconsole
@@ -122,9 +123,11 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.17.3
+jsonschema==4.18.0
     # via napari (setup.cfg)
-jupyter-client==8.2.0
+jsonschema-specifications==2023.6.1
+    # via jsonschema
+jupyter-client==8.3.0
     # via
     #   ipykernel
     #   qtconsole
@@ -137,7 +140,7 @@ kiwisolver==1.4.4
     # via
     #   matplotlib
     #   vispy
-lazy-loader==0.2
+lazy-loader==0.3
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -145,7 +148,7 @@ lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
-lxml==4.9.2
+lxml==4.9.3
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
@@ -153,7 +156,7 @@ markdown-it-py==3.0.0
     # via rich
 markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.1
+matplotlib==3.7.2
     # via napari (setup.cfg)
 matplotlib-inline==0.1.6
     # via
@@ -188,7 +191,7 @@ npe2==0.7.0
     #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.25.0
+numpy==1.25.1
     # via
     #   contourpy
     #   dask
@@ -248,7 +251,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.2 ; python_version >= "3.9"
+pandas==2.0.3 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -260,7 +263,7 @@ pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==9.5.0
+pillow==10.0.0
     # via
     #   imageio
     #   matplotlib
@@ -268,7 +271,7 @@ pillow==9.5.0
     #   scikit-image
 pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.7.0
+platformdirs==3.8.1
     # via
     #   jupyter-core
     #   pooch
@@ -279,7 +282,7 @@ pooch==1.7.0
     # via
     #   napari (setup.cfg)
     #   scikit-image
-prompt-toolkit==3.0.38
+prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
@@ -295,7 +298,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.9
+pydantic==1.10.11
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -312,7 +315,7 @@ pyopengl==3.1.6
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-pyparsing==3.1.0
+pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -328,8 +331,6 @@ pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
-pyrsistent==0.19.3
-    # via jsonschema
 pyside2==5.13.2 ; python_version != "3.8"
     # via napari (setup.cfg)
 pyside6==6.4.2 ; python_version >= "3.10"
@@ -342,7 +343,7 @@ pyside6-essentials==6.4.2
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.2
+pytest==7.4.0
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -395,6 +396,10 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
+referencing==0.29.1
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
 requests==2.31.0
     # via
     #   pooch
@@ -404,9 +409,13 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
+rpds-py==0.8.10
+    # via
+    #   jsonschema
+    #   referencing
 scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.10.1 ; python_version >= "3.9"
+scipy==1.11.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -450,11 +459,11 @@ superqt==0.4.1
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.37
+tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-tifffile==2023.4.12
+tifffile==2023.7.4
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -486,7 +495,7 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.6.3
+typing-extensions==4.7.1
     # via
     #   app-model
     #   magicgui
@@ -503,7 +512,7 @@ urllib3==2.0.3
     # via requests
 virtualenv==20.23.1
     # via napari (setup.cfg)
-vispy==0.12.2
+vispy==0.13.0
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -519,11 +528,11 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.5.0
+xarray==2023.6.0
     # via napari (setup.cfg)
 zarr==2.15.0
     # via napari (setup.cfg)
-zipp==3.15.0
+zipp==3.16.0
     # via importlib-metadata

 # The following packages are considered to be unsafe in a requirements file:
diff --git a/resources/constraints/constraints_py3.8.txt b/resources/constraints/constraints_py3.8.txt
index 884ca141..4e82451a 100644
--- a/resources/constraints/constraints_py3.8.txt
+++ b/resources/constraints/constraints_py3.8.txt
@@ -20,6 +20,7 @@ attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
+    #   referencing
 babel==2.12.1
     # via
     #   napari (setup.cfg)
@@ -34,21 +35,21 @@ certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.4
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.5
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
@@ -66,7 +67,7 @@ docutils==0.17.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.1
+exceptiongroup==1.1.2
     # via
     #   hypothesis
     #   pytest
@@ -74,16 +75,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.4
+fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.5.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,31 +92,32 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.75.3
+hypothesis==6.81.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.1
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.8.0
     # via
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==5.12.0
+importlib-resources==6.0.0
     # via
     #   jsonschema
+    #   jsonschema-specifications
     #   matplotlib
-in-n-out==0.1.7
+in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.1
+ipykernel==6.24.0
     # via
     #   napari-console
     #   qtconsole
@@ -133,13 +135,15 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.17.3
+jsonschema==4.18.0
     # via napari (setup.cfg)
-jupyter-client==8.2.0
+jsonschema-specifications==2023.6.1
+    # via jsonschema
+jupyter-client==8.3.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -148,23 +152,23 @@ kiwisolver==1.4.4
     # via
     #   matplotlib
     #   vispy
-lazy-loader==0.2
+lazy-loader==0.3
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.3
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
-lxml==4.9.2
+lxml==4.9.3
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.1
+matplotlib==3.7.2
     # via napari (setup.cfg)
 matplotlib-inline==0.1.6
     # via
@@ -183,7 +187,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -192,10 +198,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.24.4
     # via
     #   contourpy
     #   dask
@@ -255,7 +263,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version < "3.9"
+pandas==2.0.3 ; python_version < "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -267,34 +275,34 @@ pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==9.5.0
+pillow==10.0.0
     # via
     #   imageio
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21
+pint==0.21.1
     # via napari (setup.cfg)
 pkgutil-resolve-name==1.3.10
     # via jsonschema
-platformdirs==3.5.1
+platformdirs==3.8.1
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
     #   napari (setup.cfg)
     #   scikit-image
-prompt-toolkit==3.0.38
+prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -304,7 +312,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.11
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -318,7 +326,9 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
@@ -329,14 +339,12 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
-pyrsistent==0.19.3
-    # via jsonschema
 pyside2==5.15.2.1 ; python_version == "3.8"
     # via napari (setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
@@ -349,7 +357,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.4.0
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -357,13 +365,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -387,7 +395,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -401,20 +409,29 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.30.0
+referencing==0.29.1
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.5
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+rpds-py==0.8.10
+    # via
+    #   jsonschema
+    #   referencing
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.9.1 ; python_version < "3.9"
+scipy==1.10.1 ; python_version < "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -455,11 +472,14 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
+    #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
-tifffile==2023.4.12
+tensorstore==0.1.40
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+tifffile==2023.7.4
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -497,7 +517,7 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.7.1
     # via
     #   app-model
     #   ipython
@@ -511,11 +531,11 @@ typing-extensions==4.5.0
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.2
+urllib3==2.0.3
     # via requests
-virtualenv==20.23.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
-vispy==0.12.2
+vispy==0.13.0
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -533,15 +553,17 @@ wrapt==1.15.0
     # via napari (setup.cfg)
 xarray==2023.1.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
-zipp==3.15.0
+zipp==3.16.0
     # via
     #   importlib-metadata
     #   importlib-resources

 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9.txt b/resources/constraints/constraints_py3.9.txt
index ffacc9bf..696797fd 100644
--- a/resources/constraints/constraints_py3.9.txt
+++ b/resources/constraints/constraints_py3.9.txt
@@ -20,6 +20,7 @@ attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
+    #   referencing
 babel==2.12.1
     # via
     #   napari (setup.cfg)
@@ -34,25 +35,25 @@ certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.4
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.5
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.5.0
+dask==2023.7.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -66,7 +67,7 @@ docutils==0.17.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.1
+exceptiongroup==1.1.2
     # via
     #   hypothesis
     #   pytest
@@ -74,16 +75,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.4
+fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.5.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,33 +92,33 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.75.3
+hypothesis==6.81.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.1
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.8.0
     # via
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==5.12.0
+importlib-resources==6.0.0
     # via matplotlib
-in-n-out==0.1.7
+in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.1
+ipykernel==6.24.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.13.2
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -131,13 +132,15 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.17.3
+jsonschema==4.18.0
     # via napari (setup.cfg)
-jupyter-client==8.2.0
+jsonschema-specifications==2023.6.1
+    # via jsonschema
+jupyter-client==8.3.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -146,23 +149,23 @@ kiwisolver==1.4.4
     # via
     #   matplotlib
     #   vispy
-lazy-loader==0.2
+lazy-loader==0.3
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.3
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
-lxml==4.9.2
+lxml==4.9.3
     # via napari (setup.cfg)
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.1
+matplotlib==3.7.2
     # via napari (setup.cfg)
 matplotlib-inline==0.1.6
     # via
@@ -181,7 +184,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -190,10 +195,12 @@ networkx==3.1
     #   scikit-image
     #   torch
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
-numpy==1.24.3
+numpy==1.25.1
     # via
     #   contourpy
     #   dask
@@ -253,7 +260,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.3 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   xarray
@@ -265,32 +272,32 @@ pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==9.5.0
+pillow==10.0.0
     # via
     #   imageio
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.5.1
+platformdirs==3.8.1
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
     #   napari (setup.cfg)
     #   scikit-image
-prompt-toolkit==3.0.38
+prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -300,7 +307,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.11
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -314,7 +321,9 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
@@ -325,14 +334,12 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
-pyrsistent==0.19.3
-    # via jsonschema
 pyside2==5.15.2.1 ; python_version != "3.8"
     # via napari (setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
@@ -345,7 +352,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.4.0
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -353,13 +360,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -381,7 +388,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -395,20 +402,29 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.30.0
+referencing==0.29.1
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
+requests==2.31.0
     # via
     #   pooch
     #   sphinx
-rich==13.3.5
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+rpds-py==0.8.10
+    # via
+    #   jsonschema
+    #   referencing
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scipy==1.9.1 ; python_version >= "3.9"
+scipy==1.11.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -449,11 +465,14 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
+    #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
-tifffile==2023.4.12
+tensorstore==0.1.40
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
+tifffile==2023.7.4
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -491,12 +510,13 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.7.1
     # via
     #   app-model
     #   ipython
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
@@ -504,11 +524,11 @@ typing-extensions==4.5.0
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.2
+urllib3==2.0.3
     # via requests
-virtualenv==20.23.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
-vispy==0.12.2
+vispy==0.13.0
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -524,17 +544,19 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.6.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
-zipp==3.15.0
+zipp==3.16.0
     # via
     #   importlib-metadata
     #   importlib-resources

 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/constraints/constraints_py3.9_examples.txt b/resources/constraints/constraints_py3.9_examples.txt
index 1639bef1..384cdc65 100644
--- a/resources/constraints/constraints_py3.9_examples.txt
+++ b/resources/constraints/constraints_py3.9_examples.txt
@@ -20,6 +20,7 @@ attrs==23.1.0
     # via
     #   hypothesis
     #   jsonschema
+    #   referencing
 babel==2.12.1
     # via
     #   napari (setup.cfg)
@@ -34,25 +35,25 @@ certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.4
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.3
+cmake==3.26.4
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.0.7
+contourpy==1.1.0
     # via matplotlib
-coverage==7.2.5
+coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.5.0
+dask==2023.7.0
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -66,7 +67,7 @@ docutils==0.17.1
     # via sphinx
 entrypoints==0.4
     # via numcodecs
-exceptiongroup==1.1.1
+exceptiongroup==1.1.2
     # via
     #   hypothesis
     #   pytest
@@ -74,16 +75,16 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.0
+filelock==3.12.2
     # via
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.39.4
+fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.5.0
+fsspec==2023.6.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -91,33 +92,33 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.75.3
+hypothesis==6.81.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.28.1
+imageio==2.31.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.6.0
+importlib-metadata==6.8.0
     # via
     #   dask
     #   jupyter-client
     #   sphinx
-importlib-resources==5.12.0
+importlib-resources==6.0.0
     # via matplotlib
-in-n-out==0.1.7
+in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.23.2
+ipykernel==6.24.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.13.2
+ipython==8.14.0
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -131,17 +132,19 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-joblib==1.2.0
+joblib==1.3.1
     # via
     #   nilearn
     #   scikit-learn
-jsonschema==4.17.3
+jsonschema==4.18.0
     # via napari (setup.cfg)
-jupyter-client==8.2.0
+jsonschema-specifications==2023.6.1
+    # via jsonschema
+jupyter-client==8.3.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.0
+jupyter-core==5.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -150,25 +153,25 @@ kiwisolver==1.4.4
     # via
     #   matplotlib
     #   vispy
-lazy-loader==0.2
+lazy-loader==0.3
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.3
+lit==16.0.6
     # via triton
 locket==1.0.0
     # via partd
-lxml==4.9.2
+lxml==4.9.3
     # via
     #   napari (setup.cfg)
     #   nilearn
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==2.2.0
+markdown-it-py==3.0.0
     # via rich
-markupsafe==2.1.2
+markupsafe==2.1.3
     # via jinja2
-matplotlib==3.7.1
+matplotlib==3.7.2
     # via napari (setup.cfg)
 matplotlib-inline==0.1.6
     # via
@@ -187,7 +190,9 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-svg==0.1.6
+napari-plugin-manager==0.1.0a2
+    # via napari (setup.cfg)
+napari-svg==0.1.10
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -200,7 +205,9 @@ nibabel==5.1.0
 nilearn==0.10.1
     # via -r resources/constraints/version_denylist_examples.txt
 npe2==0.7.0
-    # via napari (setup.cfg)
+    # via
+    #   napari (setup.cfg)
+    #   napari-plugin-manager
 numcodecs==0.11.0
     # via zarr
 numpy==1.23.5
@@ -270,7 +277,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.1 ; python_version >= "3.9"
+pandas==2.0.3 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   nilearn
@@ -283,32 +290,32 @@ pexpect==4.8.0
     # via ipython
 pickleshare==0.7.5
     # via ipython
-pillow==9.5.0
+pillow==10.0.0
     # via
     #   imageio
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21
+pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.5.1
+platformdirs==3.8.1
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.0.0
+pluggy==1.2.0
     # via pytest
 pooch==1.7.0
     # via
     #   napari (setup.cfg)
     #   scikit-image
-prompt-toolkit==3.0.38
+prompt-toolkit==3.0.39
     # via ipython
 psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.0
+psygnal==0.9.1
     # via
     #   app-model
     #   magicgui
@@ -318,7 +325,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.7
+pydantic==1.10.11
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -332,7 +339,9 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via napari (setup.cfg)
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
@@ -343,14 +352,12 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.0
+pyqt6==6.5.1
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.0
+pyqt6-qt6==6.5.1
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
-pyrsistent==0.19.3
-    # via jsonschema
 pyside2==5.15.2.1 ; python_version != "3.8"
     # via napari (setup.cfg)
 pyside6==6.3.1 ; python_version < "3.10"
@@ -363,7 +370,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.1
+pytest==7.4.0
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -371,13 +378,13 @@ pytest==7.3.1
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.0.0
+pytest-cov==4.1.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==2.0.4
+pytest-metadata==3.0.0
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -399,7 +406,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.0.2
+pyzmq==25.1.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -413,23 +420,32 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
+    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.30.0
+referencing==0.29.1
+    # via
+    #   jsonschema
+    #   jsonschema-specifications
+requests==2.31.0
     # via
     #   nilearn
     #   pooch
     #   sphinx
-rich==13.3.5
+rich==13.4.2
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.20.0
+rpds-py==0.8.10
+    # via
+    #   jsonschema
+    #   referencing
+scikit-image==0.21.0
     # via napari (setup.cfg)
-scikit-learn==1.2.2
+scikit-learn==1.3.0
     # via nilearn
-scipy==1.9.1 ; python_version >= "3.9"
+scipy==1.11.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   nilearn
@@ -472,13 +488,16 @@ superqt==0.4.1
     # via
     #   magicgui
     #   napari (setup.cfg)
+    #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.36
-    # via napari (setup.cfg)
+tensorstore==0.1.40
+    # via
+    #   -r resources/constraints/version_denylist.txt
+    #   napari (setup.cfg)
 threadpoolctl==3.1.0
     # via scikit-learn
-tifffile==2023.4.12
+tifffile==2023.7.4
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -516,12 +535,13 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.5.0
+typing-extensions==4.7.1
     # via
     #   app-model
     #   ipython
     #   magicgui
     #   napari (setup.cfg)
+    #   pint
     #   psygnal
     #   pydantic
     #   superqt
@@ -529,11 +549,11 @@ typing-extensions==4.5.0
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.2
+urllib3==2.0.3
     # via requests
-virtualenv==20.23.0
+virtualenv==20.23.1
     # via napari (setup.cfg)
-vispy==0.12.2
+vispy==0.13.0
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -549,17 +569,19 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.4.2
+xarray==2023.6.0
     # via napari (setup.cfg)
-zarr==2.14.2
+zarr==2.15.0
     # via napari (setup.cfg)
-zipp==3.15.0
+zipp==3.16.0
     # via
     #   importlib-metadata
     #   importlib-resources

 # The following packages are considered to be unsafe in a requirements file:
-setuptools==67.7.2
+pip==23.1.2
+    # via napari-plugin-manager
+setuptools==68.0.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
diff --git a/resources/requirements_mypy.txt b/resources/requirements_mypy.txt
index 88dcaefe..285c1669 100644
--- a/resources/requirements_mypy.txt
+++ b/resources/requirements_mypy.txt
@@ -2,13 +2,13 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --output-file=resources/requirements_mypy.txt --resolver=backtracking resources/requirements_mypy.in
+#    pip-compile --config=pyproject.toml --output-file=resources/requirements_mypy.txt --resolver=backtracking resources/requirements_mypy.in
 #
 appdirs==1.4.4
     # via npe2
 build==0.10.0
     # via npe2
-click==8.1.3
+click==8.1.4
     # via typer
 docstring-parser==0.15
     # via magicgui
@@ -18,7 +18,7 @@ markdown-it-py==3.0.0
     # via rich
 mdurl==0.1.2
     # via markdown-it-py
-mypy==1.4.0
+mypy==1.4.1
     # via -r resources/requirements_mypy.in
 mypy-extensions==1.0.0
     # via
@@ -26,7 +26,7 @@ mypy-extensions==1.0.0
     #   psygnal
 npe2==0.7.0
     # via -r resources/requirements_mypy.in
-numpy==1.25.0
+numpy==1.25.1
     # via -r resources/requirements_mypy.in
 packaging==23.1
     # via
@@ -37,7 +37,7 @@ psygnal==0.9.1
     # via
     #   magicgui
     #   npe2
-pydantic==1.10.9
+pydantic==1.10.11
     # via
     #   -r resources/requirements_mypy.in
     #   npe2
@@ -66,11 +66,11 @@ types-pyyaml==6.0.12.10
     # via -r resources/requirements_mypy.in
 types-requests==2.31.0.1
     # via -r resources/requirements_mypy.in
-types-setuptools==68.0.0.0
+types-setuptools==68.0.0.1
     # via -r resources/requirements_mypy.in
 types-urllib3==1.26.25.13
     # via types-requests
-typing-extensions==4.6.3
+typing-extensions==4.7.1
     # via
     #   magicgui
     #   mypy
diff --git a/setup.cfg b/setup.cfg
index 29a65497..cd82241d 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -78,14 +78,16 @@ install_requires =
     toolz>=0.10.0
     tqdm>=4.56.0
     typing_extensions>=4.2.0
-    vispy>=0.12.1,<0.13
+    vispy>=0.13.0,<0.14
     wrapt>=1.11.1

+
 [options.package_data]
 * = *.pyi
 napari_builtins =
     builtins.yaml

+
 # for explanation of %(extra)s syntax see:
 # https://github.com/pypa/setuptools/issues/1260#issuecomment-438187625
 # this syntax may change in the future
@@ -166,6 +168,7 @@ build =
     ruff
     pyqt5

+
 [options.entry_points]
 console_scripts =
     napari = napari.__main__:main
@@ -213,6 +216,7 @@ layers=
     napari.qt
     napari.layers

+
 [importlinter:contract:3]
 name = "Block import from qt module in abstract ones"
 type = layers
--
2.34.1
